source/MaterialXGenShader/Nodes/Convolution.cpp

```diff
@@ -11,6 +11,56 @@ Convolution::Convolution()
 {
 }
 
+void Convolution::createVariables(const ShaderNode& /*node*/, ShaderGenerator& /*shadergen*/, Shader& shader)
+{
+    // Create constant for box weights
+    const float boxWeight3x3 = 1.0f / 9.0f;
+    const float boxWeight5x5 = 1.0f / 25.0f;
+    const float boxWeight7x7 = 1.0f / 49.0f;
+    vector<float> boxWeightArray;
+    boxWeightArray.push_back(1.0f);
+    for (unsigned int i = 0; i < 9; i++)
+    {
+        boxWeightArray.push_back(boxWeight3x3);
+    }
+    for (unsigned int i = 0; i < 25; i++)
+    {
+        boxWeightArray.push_back(boxWeight5x5);
+    }
+    for (unsigned int i = 0; i < 49; i++)
+    {
+        boxWeightArray.push_back(boxWeight7x7);
+    }
+    shader.createConstant(Shader::PIXEL_STAGE, 
+        Type::FLOATARRAY, "c_box_filter_weights", EMPTY_STRING, Value::createValue<vector<float>>(boxWeightArray));
+
+    // Create constant for Gaussian weights
+    vector<float> gaussWeightArray = {
```

Comment:
Could this be a constant array instead, perhaps declared in an anonymous namespace?